app/models/user.rb

```diff
@@ -267,4 +268,12 @@ def yank_gems
       deletions.create(version: v)
     end
   end
+
+  def toxic_email_domain
+    domain             = email.split("@").last
+    toxic_domains_path = Pathname.new(Gemcutter::Application.config.toxic_domains_filepath)
+    toxic = toxic_domains_path.exist? && toxic_domains_path.readlines.grep(/^#{domain}$/).any?
```

Comment:
I worry about the impact to performance by needing to searching through an ever growing list of domains. Have you run any benchmark tests to see what the performance cost is? Is there an alternative approach that doesn't share the same performance cost?

Reply:
grep takes 10ms or less.
```
puts Benchmark.measure { toxic_domains_path.readlines.grep(/^#{domain}$/).any? }
  0.010157   0.000000   0.010157 (  0.010143)
```
10ms seems like too small a variance to make a measurable impact. benchmark-ips reported version without email check was 1.03x slower https://gist.github.com/sonalkr132/fcfce8b95c796365da31597bc9d345e8

Reply:
Also to notice registration endpoint could be actually benefiting from not being fast. Current overhead is acceptable IMHO. Anyway `Rails.cache` could be used to prevent reading the file on each registration.